src/app/components/news/news.component.ts

```diff
@@ -1,25 +1,23 @@
-import { Component, OnInit, Inject, Input } from '@angular/core';
-import { NewsService, ActualNewsService } from '../../services/news.service';
+import { Component, OnInit, Inject, OnDestroy } from '@angular/core';
+import { NewsService } from '../../services/news.service';
 import { News } from '../../models/News';
 
 @Component({
   selector: 'app-news',
   templateUrl: './news.component.html',
   styleUrls: ['./news.component.scss'],
-  // Here we choose which news service is used
-  providers: [{ provide: 'NewsService', useClass: ActualNewsService }]
 })
-export class NewsComponent implements OnInit {
+export class NewsComponent implements OnInit, OnDestroy {
   news: News[];
 
-  constructor(@Inject('NewsService') newsService: NewsService) {
-    newsService.getNews().subscribe(
-      news => {
-        this.news = news;
-      });
-  }
+  constructor(@Inject('NewsService') private newsService: NewsService) {}
```

Comment:
Not sure if you'd want to address this here or in the other PR (#16), but [this is the same comment I wrote there](https://github.com/SL-edi/covid-visualisation/pull/16#discussion_r470235231)

> I could be wrong but could you remove the providers line in the metadata and have just
@Inject(ActualNewsService) newService: NewsService as the argument?
>
>Also I had come across a [InjectableToken](https://angular.io/api/core/InjectionToken) when looking up other stuff which I think is what they intend for people to use instead of just a string (for 'NewService').

Reply:
The problem with `@Inject(ActualNewsService)` is that you're no longer coding to the interface. If you wanted to use a different news service, you would have to change both `app.module.ts` and here.
Like this we handle all dependency injection through `app.module.ts`.

Reply:
If you use `@Inject(ActualNewsService)` though, you wouldn't need the provider metadata in `app.module.ts`. 
And you could still have `@Inject(ActualNewsService) private newsService: NewsService` as the constructor for the component.

The difference would be where you're telling angular what to provide the newsService attribute.

I can see us going either way (defining what to provide in `app.module.ts` or in the `@Inject...` decorator and reducing provider metadata declaration) but If we go with the way you've done it, I'll change it in #19 too for consistency.

Actually might have to specify provider metadata anyways in #19. And we'll probably want to do something similar to the newsService if we have fallback apis.

Reply:
So thinking about it a bit and I think i now get where you're coming from and its the better way. 
This way if we ever have another component which needs a `NewsService` then we'd only ever have to change the provider in `app.module.ts` right?

Still think we should use InjectableToken though

Reply:
Agreed. The only question is where to put the token's definition (i.e. which file).

Reply:
I'd say wherever we create the interface